src/main/lang/Nix.flex

```diff
@@ -136,29 +143,8 @@ MCOMMENT=\/\*([^*]|\*[^\/])*\*\/
 
   {SCOMMENT}            { return SCOMMENT; }
   {MCOMMENT}            { return MCOMMENT; }
-  {WHITE_SPACE}         { return com.intellij.psi.TokenType.WHITE_SPACE; }
-  [^]                   { return com.intellij.psi.TokenType.BAD_CHARACTER; }
-}
-
-<STRING> {
-  [^\$\"\\]+            { return STR; }
-  "$"|"$$"|\\           { return STR; }
-  \\{ANY}               { return STR_ESCAPE; }
-  "$"/"{"               { pushState(ANTIQUOTATION_START); return DOLLAR; }
-  \"                    { popState(); return STRING_CLOSE; }
 }
 
-<IND_STRING> {
-  [^\$\']+              { return IND_STR; }
-  "$"|"$$"|"'"          { return IND_STR; }
-  "''$"|"'''"           { return IND_STR_ESCAPE; }
-  "''"\\{ANY}           { return IND_STR_ESCAPE; }
-  "$"/"{"               { pushState(ANTIQUOTATION_START); return DOLLAR; }
-  "''"                  { popState(); return IND_STRING_CLOSE; }
-}
-
-<ANTIQUOTATION_START> {
-  // '$' and '{' must be two separate tokens to make NixBraceMatcher work
-  // correctly with Grammar-Kit.
-  "{"                   { popState(); pushState(YYINITIAL); return LCURLY; }
-}
+// matched by all %state states
+{WHITE_SPACE}           { return com.intellij.psi.TokenType.WHITE_SPACE; }
+[^]                     { return com.intellij.psi.TokenType.BAD_CHARACTER; }
```

Comment:
nit: I am not sure if I like these two rules being enabled in all states. If we return `TokenType.WHITE_SPACE` in a string, this would be a bug (as far as I understand it). Fortunately, the first rule of `STRING` (`[^\$\"\\]+`) and `IND_STRING` (`[^\$\']+`) would always match for at least the same length, so we will never run into this case right now. Anyway, this looks like it could cause a bug in the future if you make slight changes to the string rules. Anyway, such bug would probably be detected by the tests.

We can keep `BAD_CHARACTER` at the bottom, but I think `YYINITIAL`, `BLOCK`, and `ANTIQUOTATION` are the only states actually affected. In other states, all characters would already match other rules.

Reply:
I'll move the whitespace rule, but we should keep BAD_CHARACTER. JetBrains lexers expect that every character is covered by a token, which is `BAD_CHARACTER` as a fallback.

Reply:
Fixed by be9ec074dedfd2b99661c29e928b7477dd00e046